execution-engine/contracts/hdac-system/pop/src/pop_contract.rs

```diff
@@ -115,6 +120,45 @@ impl ProofOfProfessionContract {
 
         Ok(())
     }
+
+    pub fn vote(&self, user: PublicKey, dapp: Key, amount: U512) -> Result<()> {
+        // staked balance check
+        if amount.is_zero() {
+            return Err(Error::BondTooSmall);
+        }
+
+        // check validator's staked token amount
+        let stakes: Stakes = ContractStakes::read()?;
+        // if an user has no staked amount, he cannot do anything
+        let staked_balance: U512 = *stakes
+            .0
+            .get(&user)
+            .unwrap_or_revert_with(Error::StakesNotFound);
```

Comment:
How about this? then we can handle it in a consistent way.

```rust
let staked_balance: U512 = match stakes.0.get(&user) {
    Some(balance) => *balance,
    None => return Err(Error::StakesNotFound),
};
```

Reply:
https://github.com/hdac-io/CasperLabs/pull/14/commits/9464b5de93c7bca32387ba684abb64aee0814411